spec/models/quiz_spec.rb

```diff
@@ -6,38 +6,59 @@
   it { should validate_presence_of(:name) }
   it { should validate_uniqueness_of(:name) }
 
-  describe "#completed_by_student?" do
-    it "returns true if it has been completed by a student" do
-      user = FactoryGirl.create(:user)
-      quiz = FactoryGirl.create(:quiz_with_questions)
-      question = quiz.questions.first
-      FactoryGirl.create(:answer_submission,
-        user: user,
-        answer: question.correct_answer
-      )
-
-      expect(quiz.completed_by_student?(user)).to eq(true)
+  context "Student fills out a quiz" do
+
+    describe "#completed_at" do
+      it "returns student's completion date of quiz" do
+        user = FactoryGirl.create(:user)
+        quiz = FactoryGirl.create(:quiz_with_questions)
+        question = quiz.questions.first
+        FactoryGirl.create(:answer_submission,
+          user: user,
+          answer: question.correct_answer
+        )
+        grade = Grade.grade_user(quiz, user)
+
+        time = Time.now.round
+        expect(quiz.completed_at(user).round).to eq(time)
+      end
     end
 
-    it "returns false if it has not been completed by a student" do
-      user = FactoryGirl.create(:user)
-      quiz = FactoryGirl.create(:quiz_with_questions)
+    describe "#completed_by_student?" do
+      it "returns true if it has been completed by a student" do
+        user = FactoryGirl.create(:user)
+        quiz = FactoryGirl.create(:quiz_with_questions)
```

Comment:
We can extract the assignment of user and quiz into `let` statements since we are using them in every test.